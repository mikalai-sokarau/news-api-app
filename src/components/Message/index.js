import React from 'react';
import StyledMessage from './style';

const Message = ({ text, buttonClickHandler, buttonName }) => (
  <StyledMessage>
    <span>{text}</span>
    <a onClick={buttonClickHandler} className="waves-effect waves-light btn-small">
      {buttonName}
    </a>
  </StyledMessage>
);

export default Message;
