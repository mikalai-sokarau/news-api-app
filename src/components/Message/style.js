import styled from "styled-components";

const StyledMessage = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 50%;

  span {
    font-size: 1.3rem;
    margin-right: 10px;
  }
`;

export default StyledMessage;
