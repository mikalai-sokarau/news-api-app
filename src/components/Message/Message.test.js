import React from 'react';
import { shallow } from 'enzyme';
import Message from '../Message';

describe('<Message/>', () => {
  test('render a Message', () => {
    expect(shallow(<Message />)).toMatchSnapshot();
  });
});
