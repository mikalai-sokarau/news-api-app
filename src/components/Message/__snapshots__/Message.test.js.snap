// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message/> render a Message 1`] = `
<styled.div>
  <span />
  <a
    className="waves-effect waves-light btn-small"
  />
</styled.div>
`;
